### PR TITLE
[cli] Extend the CLI to be able to return report details

### LINKED ISSUE #2068: Extend the CLI to be able to return the bug path.
Extend the CLI to be able to return the bug path.
`CodeChecker cmd results -o json ... ` should return the bug path too.

We should consider to introduce an extra flag to return this extra information or just extend the current data.

NOTE: it will increase the number of the API calls (and DB queries), for each report we will query the bug path separately, for a large amount of reports this can mean an extra load on the server.

### PR DESCRIPTION
**Pull request by csordasmarton**
> Closes #2068

### CODE REVIEW

File: web/server/codechecker_server/api/report_server.py [hunk 1]
```diff
@@ -947,6 +1003,13 @@ def getRunResults(self, run_ids, limit, offset, sort_types,
 
                 q = q.limit(limit).offset(offset)
 
+                report_details = {}
+
+                # Get report details if it is required.
+                if get_details:
+                    report_ids = [r[1] for r in q]
+                    report_details = self.__get_report_details(report_ids)
```
Comment (gyorb) [inline web/server/codechecker_server/api/report_server.py]: Couldn't we we use the already created session to the report details?

If getting details were done in the for loop below we could get the details for the unique results too right?

Comment (csordasmarton) [reply web/server/codechecker_server/api/report_server.py]: We are getting uniqued reports in a different way, see the code above:
https://github.com/Ericsson/codechecker/blob/303cac9af72527a107cbdad8b3526a09ef1a252d/web/server/codechecker_server/api/report_server.py#L907-L921

I think for uniqued reports it doesn't make any sense to get the details.

Comment (gyorb) [reply web/server/codechecker_server/api/report_server.py]: I still do not understand why it does not make sense to get the report details for the unique reports?
It is possible that I want to know the bugpath for a unique report.

https://github.com/Ericsson/codechecker/blob/52cbd0fc7ef7a3109e5540cd3c1c46ed48d1dbc7/web/server/codechecker_server/api/report_server.py#L907-L921

Calling `report_details = get_report_details(session, [report_id])` to get the report details in the loop could be done for the unique reports too.

File: web/server/codechecker_server/api/report_server.py [hunk 2]
```diff
@@ -947,6 +1003,13 @@ def getRunResults(self, run_ids, limit, offset, sort_types,
 
                 q = q.limit(limit).offset(offset)
 
+                report_details = {}
+
+                # Get report details if it is required.
+                if get_details:
+                    report_ids = [r[1] for r in q]
```
Comment (gyorb) [inline web/server/codechecker_server/api/report_server.py]: We are querying a lot of information here and we keep the report_id only.
In the next loop we query the same information again.

Would it be much slower if we would query the report details for each report (one by one) in the for loop below?

**Pull request merged**